Error with "allowJS" and "outDir": undefined

If tsconfig has `"allowJS": true` and there are js files on emit stage ts will fail with error
`error TS5055: Cannot write file` cause it seems that it check fs before emit.

One of the workaround for that problem is to be able to give "outDir" manually in this line:
https://github.com/gravity-ui/gulp-utils/blob/main/src/createTypescriptProject.ts#L99

Gulp config with example:
https://github.com/gravity-ui/dashkit/blob/next/gulpfile.js

--- a/src/createTypescriptProject.ts
+++ b/src/createTypescriptProject.ts
@@ -1,3 +1,4 @@
+import * as os from 'node:os';
 import * as path from 'node:path';
 
 import PluginError from 'plugin-error';
@@ -96,7 +97,7 @@ export async function createTypescriptProject(options: Options = {}) {
                     rootNames: [...inputFiles.keys()],
                     options: {
                         ...tsParsedConfig.options,
-                        outDir: undefined,
+                        outDir: os.tmpdir(),
                         sourceMap: hasSourceMap,
                         inlineSourceMap: false,
                         inlineSources: false,
@@ -109,7 +110,7 @@ export async function createTypescriptProject(options: Options = {}) {
                     const output = new Map<string, OutputFile>();
                     const result = program.emit(
                         undefined,
-                        (fileName, text, _writeByteOrderMark, _onError, sourceFiles) => {
+                        (outputFileName, text, _writeByteOrderMark, _onError, sourceFiles) => {
                             if (!sourceFiles) {
                                 return;
                             }
@@ -130,6 +131,10 @@ export async function createTypescriptProject(options: Options = {}) {
                                 outputFile = {file};
                             }
 
+                            const fileName = path.resolve(
+                                path.dirname(originalName),
+                                path.basename(outputFileName),
+                            );
                             if (fileName.endsWith('d.ts')) {
                                 outputFile.dtsSource = {fileName, text};
                             } else if (fileName.endsWith('.d.ts.map')) {
@@ -164,7 +169,10 @@ export async function createTypescriptProject(options: Options = {}) {
                                     parsedMap.file = file.relative;
                                     const directory = path.dirname(file.path);
                                     parsedMap.sources = parsedMap.sources.map((name: string) => {
-                                        const absolute = path.resolve(directory, name);
+                                        const absolute = path.resolve(
+                                            directory,
+                                            path.basename(name),
+                                        );
                                         return path.relative(file.base, absolute);
                                     });
                                     applySourceMap(file, parsedMap);

